spark/src/test/scala/org/apache/spark/sql/delta/DescribeDeltaHistorySuite.scala

```diff
@@ -1473,3 +1478,54 @@ trait DescribeDeltaHistorySuiteBase
 
 class DescribeDeltaHistorySuite
   extends DescribeDeltaHistorySuiteBase with DeltaSQLCommandTest
+
+/**
+ * Represents a custom DSV2 catalog based on Delta tables. We exercise replacing the default Spark
+ * session catalog with this in order to make sure that analysis based on looking up tables by name
+ * in the session catalog works properly.
+ */
+class CustomDeltaCatalog extends DeltaCatalog {
+  override def createTable(
+      ident: Identifier,
+      columns: Array[org.apache.spark.sql.connector.catalog.Column],
+      partitions: Array[Transform],
+      properties: java.util.Map[String, String]): Table = {
+    CustomDeltaCatalog.tables.append(ident)
+    super.createTable(ident, columns, partitions, properties)
+  }
+
+  override def createTable(
+      ident: Identifier,
+      schema: StructType,
+      partitions: Array[Transform],
+      properties: java.util.Map[String, String]): Table = {
+    CustomDeltaCatalog.tables.append(ident)
+    super.createTable(ident, schema, partitions, properties)
+  }
+}
+
+object CustomDeltaCatalog {
+  val tables = ArrayBuffer.empty[Identifier]
+}
+
+class DescribeDeltaHistorySuiteWithCustomCatalog
+  extends DescribeDeltaHistorySuiteBase with DeltaSQLCommandTest {
+  val tblName = "tableName"
+
+  override protected def sparkConf: SparkConf = {
+    super.sparkConf
+      .set(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key,
```

Comment:
@dtenedor we need to test an external catalog instead of the session catalog.
e.g.
```
spark.sql.catalog.customer_catalog=org.apache.spark.sql.CustomerCatalog
```

And then we should be able to query
```
set catalog customer_catalog;
describe history table_1
```
or 
```
describe history customer_catalog.default.table_1
```